### ISSUE TITLE
Recycle type issue in the REE module for the scrubber

### ISSUE THREAD

**Issue by jurruela-cpu** [jkitchin/differentiable-flowsheets#161]
The scrubber in the module works fine by itself. However, when combining it with recycle flows, the types are changed, originating an error. I fixed it locally, by changing lines 169-172 in scrubbing.py to: 
`

            if eff["is_target"]:
                target_retained[elem] = 1.0 - jnp.asarray(eff["fraction_scrubbed"])
            else:
                impurity_removed[elem] = jnp.asarray(eff["fraction_scrubbed"])
`
This made the recycling in my flowsheet work, however, I am unsure if this implies other depending modules or notebooks won't anymore.

**Comment by jkitchin** [comment]
Fixed in the main branch. The issue was on lines 170 and 172 of src/difflow_ree/units/scrubbing.py, where float() was used to convert JAX arrays to Python floats. This works in eager mode but fails during JAX tracing (JIT, grad, or recycle solving) because tracers cannot be converted to concrete Python values.

The fix replaces float(...) with jnp.asarray(...), keeping values as JAX arrays so they remain traceable through recycle loops and gradient computations. All 34 REE tests pass with the change.

Thanks for the report and the suggested fix, @jurruela-cpu!
